GeneralsMD/Code/Tools/wdump/chunk_d.cpp

```diff
@@ -1052,6 +1052,70 @@ void	ChunkTableClass::List_W3D_CHUNK_PS2_SHADERS(ChunkItem * Item,CListCtrl *Lis
 	}
 }
 
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADERS(ChunkItem * Item,CListCtrl *List)
+{
+	List_Subitems(Item,List);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER(ChunkItem* Item, CListCtrl* List)
+{
+	List_Subitems(Item, List);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER_INFO(ChunkItem* Item, CListCtrl* List)
+{
+	uint8 version = *(uint8*)Item->Data;
+	W3dFXShaderInfoStruct* shaderinfo = (W3dFXShaderInfoStruct*)((uint8*)Item->Data + 1);
+	int counter = 0;
+
+	AddItem(List, counter, "Version", version);
+	AddItem(List, counter, "ShaderName", shaderinfo->ShaderName);
+	AddItem(List, counter, "Technique", shaderinfo->Technique);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER_CONSTANT(ChunkItem* Item, CListCtrl* List)
+{
+	int counter = 0;
+	uint8 *chunkdata = (uint8 *)Item->Data;
+	uint32 type = *(uint32 *)chunkdata;
+	chunkdata += 4;
+	// Includes the null terminator
+	uint32 constantstrlen = *(uint32 *)chunkdata;
+	chunkdata += 4;
+	char* constantname = (char*)chunkdata;
+	chunkdata += constantstrlen;
```

Comment:
How about put assert for `constantstrlen == strlen(constantname)+1`. And for the other one as well. This gives more confidence in correctness.

Reply:
It would be catastrophic for the code as is currently written. See:
```
chunkdata += constantstrlen;
```
If `constantstrlen` did not match the length of the actual null terminated string, then the next chunkdata read will be at a wrong location.

If the null terminated string was the real truth and not `constantstrlen`, then this would need to be rewritten as
```
chunkdata += strlen(constantname)+1;
```

Reply:
Ok.